plugins/rest/rest.go

```diff
@@ -61,7 +146,17 @@ func New(config []byte) (Client, error) {
 		return Client{}, err
 	}
 
-	return Client{config: parsedConfig}, parsedConfig.validateAndInjectDefaults()
+	tlsConfig, err := parsedConfig.validateAndInjectDefaults()
+	if err != nil {
+		return Client{}, err
+	}
+
+	return Client{
+		config: parsedConfig,
+		Client: http.Client{
+			Transport: &http.Transport{TLSClientConfig: tlsConfig},
```

Comment:
Does this need to be made conditional (based on whether the URL scheme is https)? I'm not sure of the answer. Basically, if someone tries to use insecure HTTP will this still work? They shouldn't do that but perhaps they already or they have some reason to (e.g., for test purposes).